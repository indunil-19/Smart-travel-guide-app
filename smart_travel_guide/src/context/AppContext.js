import React, {createContext, useReducer} from 'react';
import { initialState, reducer as AppReducer } from '../reducer/AppReducer';

export const AppContext =createContext()

export const AppProvider=({children})=>{
    const [state, dispatch]= useReducer(AppReducer, initialState)

    return(
        <AppContext.Provider  value={{state,dispatch}}>
                {children}
        </AppContext.Provider>
    )
}