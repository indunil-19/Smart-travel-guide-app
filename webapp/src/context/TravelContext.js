import React, {createContext, useReducer} from 'react';
import { initialState, Travelreducer } from '../reducer/TravelReducer';

const TravelContext =createContext()

const TravelProvider=({children})=>{
    const [state, dispatch]= useReducer(Travelreducer, initialState)

    return(
        <TravelContext.Provider  value={{state,dispatch}}>
                {children}
        </TravelContext.Provider>
    )
}

export  {TravelContext,  TravelProvider}
 