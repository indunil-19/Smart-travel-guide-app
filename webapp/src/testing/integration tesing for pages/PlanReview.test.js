import React from "react"
import {render,screen, cleanup,waitFor,fireEvent} from "@testing-library/react";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import { MemoryRouter,Router } from "react-router-dom";
import {TravelProvider} from "../../context/TravelContext"
import "@testing-library/jest-dom/extend-expect";
import { rest } from "msw";
import { setupServer } from "msw/node";
import 'whatwg-fetch'
import PlanReview from "../../screens/TravelPlanApp/PlanReview";
import { createMemoryHistory } from 'history'

describe('planReview page test', () => {
    const server=setupServer(
        rest.get("/user/getReview",(req,res,ctx)=>{
            return res(
                ctx.json({
                        rate:5,
                        review:"nice plan"
                        })
            )
        }),
        rest.post("/user/addReview",(req,res,ctx)=>{
            return res(
                ctx.json({
                        rate:req.body.rate,
                        review:req.body.review
                        })
            )
        })
    )
 
    beforeEach(()=>{      
    })
    beforeAll(() => server.listen());
    afterEach(() => server.resetHandlers());
    afterAll(() => server.close());
    afterEach(cleanup)

    it("should match snapshot",  () => {
        const tree = renderer.create(<MemoryRouter><TravelProvider><PlanReview/></TravelProvider></MemoryRouter>);
        expect(tree).toMatchSnapshot();
    });

    it("shoud render without crashing", ()=>{
        const div = document.createElement("div");
        ReactDOM.render(<MemoryRouter><TravelProvider><PlanReview  /></TravelProvider></MemoryRouter>,div)
    })

    it("should render review and rate if no error",()=>{
        const history = createMemoryHistory();
        render(<MemoryRouter><Router history={history}><TravelProvider><PlanReview/></TravelProvider></Router></MemoryRouter>)    
        waitFor(()=>{
            expect(screen.getByTestId("review")).toHaveTextContent("nice plan")
            expect(screen.getByTestId("rate")).toHaveLength(5)
            
        })
    })
    it("should add review if submit click", ()=>{
        const history = createMemoryHistory();
        render(<MemoryRouter><Router history={history}><TravelProvider><s/></TravelProvider></Router></MemoryRouter>) 

        
        waitFor(()=>{
            fireEvent.change(screen.getByTestId("review"), {target:{value:"test review"}})
            fireEvent.change(screen.getByTestId("rate"), {target:{value:4}})
            fireEvent.click(screen.getByTestId("submit"))
            waitFor(()=>{
                expect(screen.getByTestId("review")).toHaveTextContent("test review")
                expect(screen.getByTestId("rate")).toHaveLength(4)
            })
            
        })
    })


})
