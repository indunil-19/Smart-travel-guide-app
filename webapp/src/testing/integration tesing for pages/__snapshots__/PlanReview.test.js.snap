// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`planReview page test should match snapshot 1`] = `
Array [
  "s",
  <div
    className="css-1878w0a"
  >
    <div
      className="chakra-stack css-vy21t6"
    >
      <div
        className="css-1ezf8j2"
      >
        <div
          className="chakra-stack css-iroupl"
        >
          <h2
            className="chakra-heading css-1g360fb"
          >
            How do you feel about your travel plan????
          </h2>
          <div
            className="chakra-stack css-15sc1x5"
            data-testid="rate"
          >
            <svg
              className="chakra-icon css-gtaqdh"
              focusable={false}
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              className="chakra-icon css-gtaqdh"
              focusable={false}
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              className="chakra-icon css-gtaqdh"
              focusable={false}
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              className="chakra-icon css-gtaqdh"
              focusable={false}
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              className="chakra-icon css-gtaqdh"
              focusable={false}
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
          </div>
          <textarea
            className="chakra-textarea css-0"
            data-testid="review"
            disabled={true}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Describe your expereince"
            value=""
          />
          <button
            className="chakra-button css-5c2176"
            onClick={[Function]}
            type="button"
          >
            Edit
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
